task4.h

```diff
@@ -0,0 +1,267 @@
+#include <iostream>
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include <locale>
+
+using namespace std;
+
+class Figure{
+public:
+    virtual double CalcPerimetr() = 0;
+    /**
+ * @brief чисто виртуальная функция взятие периметра
+ *
+ */
+    virtual double CalcArea() = 0;
+    /**
+ * @brief чисто виртуальная функция взятие площади
+ *
+ */
+    virtual void Print() const = 0;
+    /**
+ * @brief чисто виртуальная функция вывода
+ *
+ */
+    virtual void Read() = 0;
+    /**
+ * @brief чисто виртуальная функция считывания фигуры
+ *
+ */
+    virtual ~Figure() = default;
+    /**
+ * @brief виртуальный дефолтный деструктор
+ *
+ */
+};
+
+
+class Circle: public Figure {
+private:
+    double _r, _P, _S;
+public:
+    Circle(double r, double P, double S);

```

Comment:
Если параметры функции не изменяются, то они явно должны быть обозначены `const`
[By Alex Allain](https://www.cprogramming.com/tutorial/const_correctness.html)
>The const keyword allows you to specify whether or not a variable is modifiable. You can use const to prevent modifications to variables and const pointers and const references prevent changing the data pointed to (or referenced).
>Const дает вам возможность более четко документировать вашу программу и фактически обеспечивать соблюдение этой документации. Ключевое слово const обеспечивает вашим пользователям гарантии, позволяющие оптимизировать производительность без угрозы повреждения их данных, обеспечивая соблюдение вашей документации. Например, константные ссылки позволяют указать, что упомянутые данные не будут изменены; это означает, что вы можете использовать константные ссылки как простой и немедленный способ повышения производительности для любой функции, которая в настоящее время принимает объекты по значению, не беспокоясь о том, что ваша функция может изменить данные. Даже если это произойдет, компилятор предотвратит компиляцию кода и предупредит вас о проблеме. С другой стороны, если бы вы не использовали константные ссылки, у вас не было бы простого способа гарантировать, что ваши данные не были изменены.